ocaml/typing/typecore.ml

```diff
@@ -3767,6 +3784,15 @@ and type_approx_aux_jane_syntax : Jane_syntax.Expression.t -> _ = function
   | Jexp_immutable_array _
   | Jexp_unboxed_constant _ -> ()
 
+and type_approx_aux_case_rhs ~rhs =
```

Comment:
I don't think we get any benefit from staging this function. Can we make it accept all arguments at once? (Besides, given that all callsites are full applications, the runtime performance is improved if the function declaration site takes syntactically the same number of arguments as the callsites.)

Reply:
Yeah, agreed that staging adds no value, removed. Learning more about function arity and application semantics in Ocaml was helpful to understanding this!